app/services/group/seed.rb

```diff
@@ -0,0 +1,52 @@
+module Group
+  class Seed
+    attr_reader :status
+    def initialize(tenant)
+      @user = ManageIQ::API::Common::Request.current.user
+      validate(tenant)
+    end
+
+    def validate(tenant)
+      account_number = ManageIQ::API::Common::Request.current.identity['identity']['account_number']
+      raise Catalog::NotAuthorized if account_number != tenant.external_tenant
+      raise Catalog::NotAuthorized unless @user.org_admin?
+
+      true
```

Comment:
@syncrou Does this function have to return anything, if it doesn't raise an exception its considered valid

Reply:
It doesn't, but normal rails validations either add an error if there is trouble or return a boolean. I just added the boolean to keep with that process.